### PR TITLE
Add management options to premission 

### PR DESCRIPTION
**Pull request by oshriromatrix**
Brady hello,
Here we want to add a feature that enables you to decide whether a list or option is editable in admin panel. We added a column to the list_options table called 'edit_options'. If the value is 0 it's not editable.

If a list/option is not editable it simply won't show up in the lists view in admin panel. 

If we insert 0 in the list's general row, the whole list won't show. If it's for a specific option, then only that option won't show.

Hope you like it! :)

thanks,
oshri

### CODE REVIEW

File: sql/5_0_0-to-5_0_1_upgrade.sql [hunk 1]
```diff
@@ -85,3 +85,10 @@
 --    arguments: none
 --    behavior: can take a long time.
 
+
+
+#IfMissingColumn list_options edit_options
+  ALTER TABLE `list_options` ADD `edit_options` TINYINT(1) NOT NULL DEFAULT '1' AFTER `subtype`;
```
Comment (bradymiller) [inline sql/5_0_0-to-5_0_1_upgrade.sql]: Remove the AFTER `subtype`

File: interface/super/edit_list.php [hunk 2]
```diff
@@ -1002,8 +1002,14 @@ function mysubmit() {
...
-    $res = sqlStatement("SELECT * FROM list_options WHERE " .
-      "list_id = '$list_id' ORDER BY seq,title");
+    /*
+     *  Add edit options to show or hide in list management
+     */
+    $res = sqlStatement("SELECT lo.* 
+                         FROM list_options as lo 
+                         right join list_options as lo2 on lo2.option_id = lo.list_id AND lo2.edit_options = 1
+                         WHERE lo.list_id = '{$list_id}' AND lo.edit_options = 1
+                         ORDER BY seq,title");
```
Comment (bradymiller) [inline interface/super/edit_list.php]: I am trying to wrap my brain around this. Does the following not get what you need here:
``````
$res = sqlStatement("SELECT * FROM list_options WHERE " .
       "list_id = '$list_id' AND edit_options = 1 ORDER BY seq,title");
``````

Comment (oshriromatrix) [inline interface/super/edit_list.php]: Here I get the parent for the children by the list_id: 
` right join list_options as lo2 on lo2.option_id = lo.list_id AND lo2.edit_options = 1
`
If the option_id equals to the list_id it means lo2 is a parent. 


Here we're setting lo as the children:
`WHERE lo.list_id = '{$list_id}' AND lo.edit_options = 1`  

This creates a query where if the parent is off then the children wont show too.

If we use the query you suggested above, the children will show even if the parent is off.

Comment (bradymiller) [inline interface/super/edit_list.php]: awesome.
I get it now; you are having the main list entry decide whether to show the list (and not allow the hiding of inidividual entries in the list; ie. it is all or nothing).
I'm going to bring this in and then add your explanation as a comment.

Comment (bradymiller) [inline interface/super/edit_list.php]: actually, you already have a comment, so no need for me to do that.

**Comment by bradymiller** [comment]
Just brought this into the codebase (with a couple comment changes). Thanks for the contribution! -brady

**Pull request merged**